Src/PerformanceCollector/Perf.Shared/QuickPulseTelemetryProcessor.cs

```diff
@@ -178,8 +178,15 @@ public void Process(ITelemetry telemetry)
             }
         }
 
-        private static ITelemetryDocument ConvertRequestToTelemetryDocument(RequestTelemetry requestTelemetry)
+        private static ITelemetryDocument ConvertRequestToTelemetryDocument(RequestTelemetry requestTelemetry, IApplicationIdProvider applicationIdProvider)
         {
+#if NET45
+            // some of the requestTelemetry properties might be deffered by using RequestTrackingTelemetryModule.DisableTrackingProperties.
+            // evaluate them now:
+            var request = System.Web.HttpContext.Current?.Request;
```

Comment:
There were come cases when HttpContext.Current is not available or lost in the process of the Http Request execution. Do any of them apply here? We simply return if Context is null - any downsides of that?

This also adds a dependency for Telemetry Processors to run synchronously with the request (as they do now) due to the use of Current, which was not the case before. If we decide to move Processors to async execution in 3.0, it may break. Arguably, 3.0 may have another approach to involve sampling earlier anyway, so this won't be needed...

Reply:
There are such potential cases with design change. The more async handling should be considered along with sampling redesign, where it happens before any expensive operations.